4.10.0 haiku

Sylvain78 opened the pull request:
In order to compile ocaml on Haiku, the network functions reside in lnetwork, and two formats had to be modified.

Octachron committed:
disambiguation for extension constructors

This commit exposes all extension constructors when looking up for
a construction with a given type in the environment.

This makes constructor disambiguation work for extension constructors.

Going one step further, when the name of an extension constructors
is misspelled, we cannot rely on the type to find all possible
names. However, since we are in an error path, we have some
path at hand.
Thus, this commit alters the "lookup_from_type" function in the
Constructor module to make it scan the whole environment for
extension constructors with the right type.

This commit should not change anything for labels and standard constructors.

trefis committed:
matching: half_compiled_row: (head, patl, act) => - 12 assert false

trefis committed:
matching: move things around

trefis committed:
matching: Add intermediary types to keep track of the compilation state of pms

trefis committed:
parmatch: export Pattern_head

trefis committed:
matching: start using Pattern_head

trefis committed:
typedtree.mli: un-open Types

trefis committed:
[REVIEW REQUIRED] matching: simplify can_group

review of can_group factorization by @gasche:

> I reviewed the change to `can_group` and believe it is correct.
> (I knew it would be nicer as a binary operation!)
>
> The different treatments of Lazy and Tuple/Record does look a bit odd,
> but I believe that it is actually the right thing to write.
>
> In the Tuple or Record case, the idea is that we can group Any heads
> with the Tuple/Record case and just explode all of them (including the
> Any cases) according to the tuple/record arity.
>
> In the Lazy case, the corresponding choice would be to add Any values
> to the Lazy group, and force all the elements of the group. This is
> not so nice, because intuitively we shouldn't force Lazy values unless
> we have to.
>
> One may expect that in general the argument of the pattern will be
> forced anyway, as long as there is at least one Lazy pattern above in
> the matrix, so it doesn't really matter whether we include the Any
> patterns in the forced group or not. I would argue that (1) even if
> that was true, it would be semantically dubious to handle Any that way
> (see a more precise criterion below), (2) I am not actually sure that
> this is true, what if the first group gets found unreachable by
> splits in following columns?
>
>     # type void = | ;;
>     # match (lazy (print_endline "foo"), (None : void option)) with
>       | lazy (), Some _ -> .
>       | _, None -> false;;
>     - : bool = false
>
> This gives the following decision tree for whether Any is included in
> the group:
>
> - Can the absence of Any be used to generate nice/efficient tests for
>   the heads of the group? In that case, don't include Any.
>   (Not included: all the Constant cases, Array (discriminate on size),
>    String, etc.)
>
> - Is Any always exactly equivalent to a more-defined head for values
>   of this type? In that case, do include Any, otherwise do not.
>   (Included: Variant, Record)
>   (Not included: Lazy)

gasche committed:
matching: what_is_cases returns a head

gasche committed:
matching: inline split_and_precompile* to clarify compile_*

gasche committed:
matching: factorize compile_match and compile_match_nonempty

gasche committed:
matching: minor code factorization of compile_* functions

trefis committed:
Add a unique id to every signature item

trefis committed:
use uid for usage warnings

trefis committed:
typecore: wrap_unpacks => type_unpacks

This allows us to give the same uid to the module bound in the guard,
and the one bound in the rhs.

trefis committed:
update testsuite

trefis committed:
set_value_used_callback: simplify implementation

trefis committed:
Changes

Gbury committed:
Make asmgen tests run

Relative paths are needed in ocamltest (rather than plain executable
file names), because else 'execvp' searches for the executable in the
PATH instead of the current directory.

gretay-js committed:
Add -function-sections to missing folder in otherlibs

gasche committed:
disambiguation: improve the API of NameChoice.disambiguate

lpw25 committed:
Short circuit simple inclusion checks

gasche committed:
Gc.Memprof.start: take a record instead of 5 optional parameters

The Gc.Memprof module provides a low-level API, that will hopefully be
paired with user libraries that provide high-level instrumentation
choices.

A natural question is: how are the higher-level API going to expose
their choice of instrumentation to their users? With the current
Memprof.start API (before this patch), they would have to either
provide their own `start` function wrapping Memprof.start, or provide
a tuple of callbacks for to their users to pass to Memprof.start
themselves.

    val start : params -> unit
    (* or *)
    val callback : params ->
      ((allocation -> foo option) * (allocation -> bar option) * ... )

With an explicit record, it is easier for libraries to expose an
instrumentation choice (possibility parametrized over
user-provided settings):

    val tracker : params -> (foo, bar) Gc.Memprof.tracker

In addition, providing a record instead of optional parameters makes
it much easier to provide "default settings" (helper functions) that
instantiates the types `'minor` and `'ḿajor`, see for example
`simple_tracker` in this patch (which stores the same information for
the minor and major heap, and does not observe promotion), or to later
define checking predicates that can verify that a given choice of
callbacks is sensible (for example: providing a major-dealloc callback
but no promotion callback (dropping all tracked value on promotion) is
probably not a good idea).

Bootstrap: to avoid requiring an awkward bootstrap, this commit keeps
the (now unused) function caml_memprof_start_byt unchanged -- it is
used in the bootstrap binaries, so removing it would break the
build. The intention is to remove it in the following commit.

gasche committed:
memprof runtime: keep tracker callbacks in a single value

lpw25 committed:
Bootstrap

lpw25 committed:
Avoid polymorphic compare in Ident

stedolan committed:
Use typing information from Clambda for mutable Cmm variables

Gbury committed:
Fix arith.cmm test

An error was triggered on the floating point computations because of
rounding errors between the read from the D array, and the C-side fp
computation which used 80-bit precision.

shindere committed:
Merge pull request #9319 from shindere/remove-useless-calls-to-cat

runtime/Makefile: Get rid of two useless calls to cat (cosmetic)

shindere committed:
Merge pull request #9152 from Gbury/ocamltest_asmgen

Make asmgen tests run

smuenzel-js committed:
move changes from 4.09 to 4.10

Octachron committed:
Merge pull request #9323 from smuenzel-js/changes-four-09

Some "Changes" in 4.09 weren't included in 4.09

gasche committed:
Merge pull request #9318 from trefis/empty_in_or_pattern

Fix exhaustivity checking in presence of empty types

Octachron committed:
Merge pull request #9308 from dbuenzli/bootdoc

BOOTSTRAP.adoc: clarify some points (closes #9278).

nojb committed:
Fix typo in Changes

Octachron committed:
Revert "Make asmgen tests run"

This reverts commit e168edc67dd963321f8460baf02b837cf8d55bca.

stedolan committed:
Assert that Cassign is only used on Clet_mut-bound variables

glennsl committed:
Document base case of List.for_all and List.exists (#9325)

* Document the base case (`[ ]`) of List.for_all and List.exists

dra27 committed:
Testsuite is not parallel

dra27 committed:
Testsuite is not parallel

dra27 committed:
Correct implementation of --disable-stdlib-manpages

Octachron committed:
Fix font family in code snippet (#9334)

dra27 committed:
tests/tool-ocamldep-modalias is not parallel

dra27 committed:
tests/tool-ocamldep-modalias is not parallel

trefis committed:
Types.Uid: include Identifiable

Octachron committed:
matching: more precise argo argument

lpw25 committed:
Add test for short-paths in error messages

lpw25 committed:
Fix some bugs in short-paths

lpw25 committed:
Apply short-paths to an additional case

Octachron committed:
env summary: don't record implicit cmi files

We don't want to record the state of the file system at the start
of the compilation in the compiled files.
Consequently, we only add persistent modules to the env summary
if they have an observable action on the initial environment.
This is only the case if they shadow a non-persistent module of the
initially opened library (which can only be Stdlib currently).

lpw25 committed:
Add Changes entry

lpw25 committed:
Add Changes entry

lpw25 committed:
Add Changes entry

lpw25 committed:
Merge pull request #9305 from lpw25/avoid-poly-compare-in-ident

Avoid polymorphic compare in Ident

trefis committed:
Merge pull request #9275 from lpw25/short-circuit-simple-inclusion-checks

Short circuit simple inclusion checks

lpw25 committed:
Add [@inlined hint] attribute

stedolan committed:
update Changes

trefis committed:
Uid.reinit

trefis committed:
printtyp: inline a function call (#9347)

trefis committed:
bootstrap

trefis committed:
add forgotten word in comment

stedolan committed:
Merge pull request #9316 from stedolan/cmm-let-mut

Use typing information from Clambda for mutable Cmm variables

trefis committed:
build_subtype: assert arrow type is known (#9348)

`build_subtype` is invoked only through `enlarge_type` when typechecking Pexp_coerce.

The type passed to `build_subtype` can only have `Cok` arrows because:
- the base type is the direct result of `Typetexp.transl_simple_type_delayed` which only introduces `Cok` arrows
- the other source of arrows could be the result of unification; however at this point the type variables are fresh, they have not been unified with any existing types.

Adding this assertion ensures that this property won't be broken by accident.

gasche committed:
make depend

sbriais committed:
Define Seq.cons and Seq.append. (#9077)

* Define Seq.cons and Seq.append.

Octachron committed:
Merge pull request #9343 from lpw25/fix-short-paths-environments

Re-enable `-short-paths` for some error messages

trefis committed:
Merge pull request #8934 from trefis/usage

Stop relying on location to track usage

gasche committed:
Merge pull request #9196 from gasche/clarify-disambiguation-3

disambiguation: improve the interface of `NameChoice.disambiguate`

maroneze committed:
Document warning 67 in ocamlc man page (#9346)

stedolan committed:
Micro-optimise allocations on amd64 to save a register (#9280)

There's no need for allocation on amd64 to clobber the %rax register. It's only used in one case (-compact out-of-line allocation of >3 words), and only used there to do a single subtraction. That subtraction can be done by the caller at no code size penalty, freeing up %rax.

Inside amd64.S functions, %r11 can be used instead of %rax as temporary.  %r11 is destroyed by PLT stub code, so on ELF platforms it costs nothing to use.

gasche committed:
fix HACKING.adoc formatting

trefis committed:
matching: remove dedicated type, use labelled arguments

trefis committed:
matching: add an assert false in divide variant

This triggers if the first column contains something other than a
variant (this is similar to the "get_key"s for other heads).

stedolan committed:
Implement #9280 for MSVC64 (#9358)

Follow-up to 4d4a056bc

gasche committed:
matching: deduplicate the matrix pretty-printing functions

gasche committed:
matching: deduplicate the debug information printing

trefis committed:
Merge pull request #9322 from trefis/rematch-structured-rows

[matching.ml cleanup] structured rows

Octachron committed:
ocamldebug: initialize all functions in Env (#9356)

Switch ocamldebug to compiler-libs in order to avoid hidden dependencies issue. 

In particular, the Env module is only fully initialized after the Typemod and Includemod modules have been linked. Calling `Env.find_type` before that initialization may result in an assert false when functor-derived types are involved.

trefis committed:
matching: ctx_matcher with heads (#9359)

Co-authored-by: Gabriel Scherer <gabriel.scherer@gmail.com>

stedolan committed:
Avoid looking up nonexistent Comballoc debuginfo

muskangarg21 committed:
TOOLS: Ocamldep arg to not include current dir to search path

nojb committed:
Deprecate -annot (#2141)

* Move driver code from Cmt2annot to Read_cmt
* Move cmt2annot.ml into typing/
* make depend
* Use standard error handling
* Move specific logic to read_cmt
* Do not pass full cmt record as argument
* Better locations
* Emit .annot files produced from cmt data
* Remove direct calls to Stypes
* Deprecate -annot
* Changes
* make depend
* Adapt doc
* make -C tools depend

gasche committed:
Map.filter_map and Set.filter_map

gasche committed:
minor Changes files

stedolan committed:
Add an assertion that we never try to read debuginfo from system frames (#9366)

stedolan committed:
Generate the same locations for raises in bytecode and native code

stedolan committed:
Generate the same backtraces for method send in bytecode and native

jeremiedimino committed:
Add a new toplevel directive #use_output "<command>"

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>
Co-authored-by: David Allsopp <david.allsopp@metastack.com>

stedolan committed:
Changes entry

gasche committed:
Merge pull request #9362 from stedolan/memprof-debuginfo-fix

Avoid looking up nonexistent Comballoc debuginfo

Octachron committed:
Merge pull request #9357 from muskangarg21/makedepend

[Ocamldep] [TOOLS] : -nocwd argument to not include current dir to search path

stedolan committed:
Merge pull request #9367 from stedolan/unify-backtraces

Make bytecode and native-code backtraces agree

jeremiedimino committed:
Fix #9344 (#9368)

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>

redianthus committed:
fix doc comment for unit type

muskangarg21 committed:
[Refactor]: Typing/typeclass.ml

Octachron committed:
Merge pull request #9373 from zapashcanon/unitdoc

fix doc comment for unit type

redianthus committed:
fix doc comment for seq type

redianthus committed:
fix sockaddr type doc comment

redianthus committed:
fix msg_flag type doc comment

Octachron committed:
Merge pull request #9379 from zapashcanon/seqdoc

fix doc comment for seq type

dra27 committed:
Use $AWK not awk

Ensures gawk is used if available.

dra27 committed:
Don't assume . in AWKPATH

Ensure all script invocations explicitly refer to a directory.

Octachron committed:
Merge pull request #9380 from zapashcanon/unixdoc

Unixdoc

dra27 committed:
Merge pull request #9383 from dra27/explicit-awk

Don't assume . in AWKPATH

trefis committed:
matching: use polymorphic variants to classify general/half_simple/simple patterns (#9361)

Also: these types are now instances of Typedtree.pattern_data.

Co-authored-by: Gabriel Scherer <gabriel.scherer@gmail.com>

xavierleroy committed:
The manual is now hosted on ocaml.org (#9378)

* Point to ocaml.org for the other versions of the manual
* Remove mentions of caml.inria.fr

ocaml.org is now the only place to go.


* MacOS X is now called macOS; MacOS 9 is no more
* Update availability of dynamic loading of C stubs
* "website" is more common these days than "Web site"

lpw25 committed:
Make substitution of type functions more robust

lpw25 committed:
Add regression test

lpw25 committed:
Prohibit signature local types with constraints

Anukriti12 committed:
call_linker now returns exit_code for better error response on linking_error, fixes #7141

stedolan committed:
Visit registers at most once in Coloring.iter_preferred

lpw25 committed:
Improve usage warnings for recursive modules

lpw25 committed:
Add test for usage warnings of recursive modules

stedolan committed:
Fix Changes entry

dra27 committed:
Merge pull request #9396 from stedolan/fix-changes-entry

Reveal the identity of a mysterious reviewer

gasche committed:
Merge pull request #9374 from muskangarg21/refactor

Refactor typing/typeclass.ml class environments

gretay-js committed:
Replace caml_int_compare and caml_float_compare with primitives (#2324)

xavierleroy committed:
extern.c: make sure extern_free_stack() is always called on an exception

In the original code, if `caml_output_value_to_block` raises an exception,
`free_extern_output` does not call `extern_free_stack` because
of the early return on `extern_userprovided_output != NULL`.

muskangarg21 committed:
Stdlib: added Printf.ikbprintf, Printf.ibprintf (#9364)

Octachron committed:
fix attribute examples in the manual

Octachron committed:
Merge pull request #9401 from Octachron/manual_attributes

manual: fix attribute examples

gasche committed:
minor Changes change

y-yu committed:
Remove sudo:false in .travis.yml (#9402)

muskangarg21 committed:
[nitpick]: Added "." at the end of warning descriptions in utils/warnings.ml

trefis committed:
dune: cmt2annot

shindere committed:
Merge pull request #9222 from gasche/fix-ocamltest-dependencies

fix Makefile dependencies of ocamltest

muskangarg21 committed:
Added: Warning 67 description.

gasche committed:
Merge pull request #9403 from muskangarg21/warn67

Added warning 67 description and nitpick for "."

Anukriti12 committed:
Merge branch 'trunk' into trunk

stedolan committed:
Changes entry

Octachron committed:
packed modtype lookup can fail due to missing cmi

This commit removes an assert false and makes the missing
cmi case behaves like the abstract case.

garrigue committed:
add optional argument principality examples (#9405)

gasche committed:
memprof: remove caml_unused memprof_start_byt and then bootstrap

Build steps for this commit (from a valid core build for the previous commit):

    make coreall
    make coreboot

(Note: `make core` would not work as we remove a primitive.)

trefis committed:
type_application: merge sargs and more_sargs

trefis committed:
apply the same treatment to Pcl_apply

trefis committed:
extract_label: return an option instead of raising

trefis committed:
add more principality warnings examples for applications (#9408)

trefis committed:
Changes

trefis committed:
typeclass: further simplification as suggested by reviewer

jeremiedimino committed:
Move changelog entry for #9344's fix in 4.10 section

Signed-off-by: Jeremie Dimino <jeremie@dimino.org>

gasche committed:
ocamltest: collect 'summary' information on whether a failure happened

gasche committed:
ocamltest: opt-in -keep-test-dir-on-success to keep test data on success

gasche committed:
testsuite/Makefile: refactor the -promote ocamltest logic

The existing approach requires to set all ocamltest flags at once on
recursive calls. This is very inconvenient if we want to set more than
one flag -- we wish to support choosing both -promote
and -keep-test-dir-on-success.

gasche committed:
testsuite/Makefile: allow users to explicit set KEEP_TEST_DIR_ON_SUCCESS=1

gasche committed:
Changes -- and justification.

On my machine, running the whole testsuite generates 1.5Gio of test
data. I have 11 workdirs for the OCaml repository (versions from 4.08
to trunk, plus several experimental workdirs); at any point in time
there can be more than a dozen gibibytes of test data on my disk, and
this is too much.

I use the test data saved by ocamltest when a test fails, to diagnose
the failure. I don't remember ever looking at the test data of
a succesful test.

The present patchset changes ocamltest to, by default, discard the
test directory of succesful tests -- the test artefacts are kept only
when there is a failure.

If one wishes to preserve the test data of succesful tests, one should
explicitly pass the -keep-test-dir-on-success flag. The
testsuite/Makefile is then changed to pass this flag if the user set
the KEEP_TEST_DIR_ON_SUCCESS variable to a non-empty value.

gasche committed:
Merge pull request #9254 from gasche/memprof-record

Gc.Memprof.start: take a record instead of 5 optional parameters

Anukriti12 committed:
code review changes

gasche committed:
Merge pull request #9414 from gasche/testsuite-drop-passed-artefacts

ocamltest+testsuite: by default, only keep test data on failure

gasche committed:
Merge pull request #9365 from gasche/Set-Map-filter_map

Map.filter_map and Set.filter_map

gasche committed:
[minor] matching: rename `group_var` into `simple_omega_like`

(Report from Florian Angeletti)

xavierleroy committed:
caml_output_value_to_malloc: revise freeing of output buffer

This commit changes caml_output_value_to_malloc to use the same
pattern as caml_output_val and caml_output_value_to_bytes:
the blocks of output are freed in the same loop that copies
them to the final destination.

Originally, caml_output_value_to_malloc calls free_extern_output
to free the blocks of output.  This is correct too, but causes
extern_free_stack to be called twice, once at the end of extern_value
and once in free_extern_output.  This is OK because extern_free_stack
is protected against double free errors.

Still, I find it more elegant, more consistent with the rest of the
code, and less error-prone w.r.t. double free errors to not call
free_extern_output at the end of caml_output_value_to_malloc.

Later, free_extern_output could be renamed to e.g. extern_finalize
to emphasize that it is to be called when something goes wrong and
we are about to raise an exception.

avsm committed:
README: update link to manual pages and reference inbox.ocaml.org

Anukriti12 committed:
replaced fib example with gcd in coreexample.etex

xavierleroy committed:
configure for mingw: GCC optimization options

Require GCC >= 5.0 and use the same GCC optimization options as for
the other ports.

Octachron committed:
Merge pull request #9423 from avsm/readme-new-site

README: update link to manual pages and reference inbox.ocaml.org

Octachron committed:
Merge pull request #1706 from Octachron/extension_disambiguated_bis

Spellchecker hints and type-directed disambiguation for extensible variants: bis

xavierleroy committed:
configure: add GCC option -fexcess-precision=standard

This option forces GCC to follow the ISO C standards concerning
rounding of intermediate FP results.  It avoids some FP issues
with the x86 32 bits ports of OCaml, which can run into
excess precision problems due to the x87 FP unit.

Closes: #7917

xavierleroy committed:
configure: quote [] inside cases of AS_CASE construct

Patterns such as `gcc-[012]-*` are expanded as `gcc-012-*`.
An extra quoting is needed: `gcc-[[012]]-*` is expanded as `gcc-[012]-*`.

gretay-js committed:
Define OPTCOMPFLAGS in Makefile.common.in

gretay-js committed:
Check FUNCTION_SECTIONS in Makefile.common.in to update OPTCOMPFLAGS

gretay-js committed:
Check FLAMBDA in Makefile.common.in to update OPTCOMPFLAGS

gasche committed:
Merge pull request #9417 from trefis/rematch-can_group-simplif

matching: simplify can_group

gasche committed:
[minor] fix a Makefile warning by defining undefined variables

Report by Xavier Leroy:
  https://github.com/ocaml/ocaml/pull/9414#issuecomment-610231289

gasche committed:
Merge pull request #9422 from Anukriti12/gcd

replaced fib example with gcd in coreexample.etex

gasche committed:
Changes entry for the refactoring of the pattern-matching compiler

xavierleroy committed:
PowerPC, System Z: do not reset backtrace_pos in caml_raise_exception (#9428)

* PowerPC, System Z: do not reset backtrace_pos in caml_raise_exception
* Add Changes entry

trefis committed:
type_args: forbid optional arguments commuting when [ignore_labels=true]

trefis committed:
type_args: remove ty_old

This was useful when commuting optional arguments was allowed with
`ignore_labels = true`. Which it isn't anymore.

JulienRouse committed:
Fix documentation comment for struct large_free_block (#9399)

Fix the name of the field `isnode` in the struct `large_free_block` in the comment documenting it (was `node` before).

Replace "son" by "child" in the documentation of the struct "large_free_block".

xavierleroy committed:
Update Changes

xavierleroy committed:
Merge pull request #9420 from xavierleroy/output-value-fixes

Minor fixes to the marshaler (output_value)

trefis committed:
type_args: some comments (and formatting noise)

trefis committed:
type_unknown_args is a fold

trefis committed:
type_application: primitives inspection helper

xavierleroy committed:
configure: use m4_normalize to reformat long messages

xavierleroy committed:
Update Changes for PR#9426

garrigue committed:
In `{expr with ...}`, always evaluate `expr` even if all labels are redefined (#9432)

This commit reverts c1a7ace (originally c545e04), which was a
temporary fix that is no longer needed because it was superseded by
#6608.

The temporary fix caused `{expr with lbl1 = e1; ... }` to not evaluate
`expr` if all labels of its type are overriden.  As reported in #7696
this is not desirable.  Reverting the temporary fix causes `expr` to
be evaluated always.

As a consequence, a corner case of value "let rec" is no longer accepted.
The corresponding test was updated.

Closes: #7696

xavierleroy committed:
Merge pull request #9426 from xavierleroy/configure-mingw

Configure: improve selection of GCC options

trefis committed:
inline type_unknown_args

trefis committed:
type_application: better names, some more comments

trefis committed:
type_application: implement ignored and omitted in the same way

trefis committed:
testsuite: -nolabels and commuting optional arguments

trefis committed:
Typeclass.type_args rewrite in the same way

xavierleroy committed:
configure: use cc as assembler with clang and for all FreeBSD platforms (#9437)

In recent FreeBSD, `cc` is Clang and `ld` is LLD, the LLVM linker, but
`as` is still GNU binutils.  Moreover, Clang contains its own
assembler and does not call `as`.  Consequently, object files produced
by invoking `as` directly are slightly different from those produced
by `cc`.

This can cause obscure errors such as issue #9068: `ld -r` fails when
combining objects produced by `as` and objects produced by `cc`.

The workaround is to use `cc` as the assembler.  We already did that
for the ARM and ARM64 targets, but #9068 shows that it is necessary
for AMD64 too.  Just use `cc` as assembler for all FreeBSD targets.

Similar issues were reported under Linux when clang and LLD are used
instead of GCC and binutils.  We already used clang as the preprocessed
assembler in this case.  Also use clang as the assembler in this case.

Closes: #9068

stedolan commented on ocamltest/run_unix.c:
```diff
@@ -258,7 +258,7 @@ static int handle_process_termination(
   corestr = core ? "" : "no ";
   fprintf(stderr,
-    "Process %d got signal %d(%s), %score dumped\n",
+    "Process %ld got signal %d(%s), %score dumped\n",
```
This isn't a portable fix - neither `%d` nor `%ld` is generally correct for printing a `pid_t`. I think the right thing is to use `%ld`, but replace the `pid` argument with `(long)pid`.

stedolan committed:
Fix tail-call optimisation with a mutable ref (#9443)

Fix tail-call optimisation with a mutable ref
(Clet_mutable was not recognized properly in tail position.)

Add a test for tail-call optimisation with a mutable ref

gadmm committed:
Install a pretty printer for the Fun.Finally_raised exception (#9266)

gasche committed:
Changes

stedolan committed:
Reset Reg.visit_generation during Reg.reset

gasche committed:
Merge pull request #9447 from trefis/rematch-more-heads

pattern-matching refactoring: keep going

lpw25 committed:
Add Changes entry

xavierleroy committed:
Reimplement output_value using a hash table to detect sharing (#9353)

The previous implementation was doing temporary in-place modifications
of the OCaml value being marshaled.  This is incompatible with
Multicore OCaml.

dra27 committed:
Merge pull request #8639 from dra27/autoconf-tweak

autoconf tweaks

lpw25 committed:
Merge pull request #9349 from lpw25/inline-hint

Add [@inlined hint] attribute

dra27 committed:
Introduce $MAKE to AppVeyor script

dra27 committed:
Introduce $MAKE to AppVeyor script

lpw25 committed:
Add locations to docstring attributes

dra27 committed:
Display AppVeyor environment

dra27 committed:
Display AppVeyor environment

dra27 committed:
Run AppVeyor with -j for branches

dra27 committed:
Run AppVeyor with -j for branches

dra27 committed:
Use configure-caches on AppVeyor

dra27 committed:
Use configure-caches on AppVeyor

dra27 committed:
Merge pull request #9335 from dra27/fix-stdlib-manpages

Correct implementation of --disable-stdlib-manpages

trefis committed:
type_application: better variable name

lpw25 committed:
Add Changes entry

dra27 committed:
Merge pull request #9057 from dra27/debugging-the-debugger

Aid debugging the debugger

gretay-js committed:
Remove stray newlines

gretay-js committed:
Add an entry to Changes

trefis committed:
testsuite: -nolabels, commutes now, might get eliminated later

Octachron committed:
Merge pull request #9409 from Octachron/yet_another_implicit_deps_error

packed modtype lookup can fail due to missing cmi

trefis committed:
optional arguments eliminated only when followed by unlabelled argument

trefis committed:
illegal commutation: more information in error message

gasche committed:
Merge pull request #9389 from Anukriti12/trunk

call_linker now returns exit_code for better user response on linking_error, fixes #7141

dra27 committed:
Fix #mod_use in toplevel

Regression from #9283.

Fixes #9455

dra27 committed:
Add a test for #mod_use

dra27 committed:
s/wrap_mode/wrap_mod/

dra27 committed:
Fix the native toplevel too...

trefis committed:
Further Changes

gasche committed:
toplevels: label the wrap_mod argument

lpw25 committed:
Merge pull request #9452 from lpw25/add-locations-to-docstring-attributes

Add locations to docstring attributes

lpw25 committed:
Add Changes entry

lpw25 committed:
Merge pull request #9387 from lpw25/more-robust-type-function-subst

Make substitution of type functions more robust

lpw25 committed:
Add Changes entry

lpw25 committed:
Merge pull request #9388 from lpw25/no-siglocal-type-constraints

Prohibit signature local types with constraints

lpw25 committed:
Merge pull request #9393 from lpw25/improve-recmodule-usage-warnings

Improve recursive module usage warnings

gasche committed:
add a regression test for issue 9460

See <https://github.com/ocaml/ocaml/issues/9460>.
We currently expect this test to fail on 32bit systems, and succeed on 64bits systems.

Contributed by Richard Jones.

lthls committed:
Fix Cmm type of unboxed Int64 values in Clet_mut

xavierleroy committed:
Turn off PIE for musl-based Linux on platforms other than amd64 and s390x (#9456)

Alpine Linux and perhaps other musl-based Linux distributions produce
position-independent executables (PIEs) by default.  If non-PIC object
files are given to the linker, it silently produces a wrong executable
that crashes when run.  This is the case for ocamlopt-generated code,
which by default is not PIC except on amd64 (x86_64) and s390x (Z systems).

Closes: #7562

gasche committed:
Changes entry

trefis committed:
Merge pull request #9411 from trefis/type_args-merge-sargs

forbid optional arguments reordering with -nolabels

nojb committed:
x86 asm: handle unit names with special characters (#9465)

nojb committed:
Fix asmcomp unit name test for bsd (#9473)

dra27 committed:
Merge pull request #9454 from dra27/faster-appveyor

AppVeyor speed-ups

dra27 committed:
Revert "AppVeyor speed-ups"

dra27 committed:
Merge pull request #9478 from ocaml/revert-9454-faster-appveyor

Revert "AppVeyor speed-ups"

dra27 committed:
Merge pull request #9479 from dra27/faster-appveyor

AppVeyor speed-ups take 2

gasche committed:
Merge pull request #9463 from lthls/fix_int64_cmm_typ

Fix Cmm type of unboxed integers in Clet_mut

xavierleroy committed:
Cast `getpid()` to `long long` before printing

Follow-up to commit 41441a65f.
That commit fixed the issue in runtime/ .
The present commit fixes the issue in ocamltest/ .

dra27 committed:
Test distclean in CI

Ensure that distclean works in an unconfigured tree. At the end of the
build, ensure that no tracked files have been altered or untracked files
created and ensure that distclean removes all ignored files.

Octachron committed:
Merge pull request #9345 from Octachron/reproducible_env

Reproducible env summaries

nojb committed:
Use Filename.quote_command (#9476)

dra27 committed:
Move Changes entry

dra27 committed:
Merge pull request #9165 from gretay-js/fs-otherlibs

Add missing -function-sections in Makefile in otherlibs

dra27 committed:
Merge pull request #9457 from dra27/fix-mod_use

Fix #mod_use in toplevel

dra27 committed:
Merge pull request #9480 from xavierleroy/print-pid_t

Cast `getpid()` to `long long` before printing

stedolan committed:
Make Cconst_symbol have typ_int to fix no-naked-pointers mode (#9282)

dra27 committed:
Merge pull request #8631 from dra27/unify--c

Use same cflags and cppflags for ocamlc -c and ocamlopt -c (#7678)

dra27 committed:
Merge pull request #9481 from dra27/check-hygiene

Test make distclean fully in CI

xavierleroy committed:
Merge pull request #9392 from stedolan/visit-once

Visit registers at most once in Coloring.iter_preferred.

Sylvain78 committed:
Compiling for Haiku : network functions reside in libnetwork

Sylvain78 commented:
Closing because the format of getpid values in run_unix.c have been changed on trunk.
